api/src/test/java/org/pmiops/workbench/api/RuntimeControllerTest.java

```diff
@@ -454,24 +480,95 @@ public void testDeleteRuntimesInProject_NullRuntimesList() {
+
+    runtimeController.createRuntime(BILLING_PROJECT_ID, null);
+    verify(userRuntimesApi).createRuntime(eq(BILLING_PROJECT_ID), eq(getRuntimeName()), any());
+  }
+
+  @Test
+  public void testCreateRuntime_emptyRuntime() throws ApiException {
+    when(userRuntimesApi.getRuntime(BILLING_PROJECT_ID, getRuntimeName()))
+        .thenThrow(new NotFoundException());
+    stubGetWorkspace(WORKSPACE_NS, WORKSPACE_ID, "test");
+
+    runtimeController.createRuntime(BILLING_PROJECT_ID, new Runtime());
+    verify(userRuntimesApi).createRuntime(eq(BILLING_PROJECT_ID), eq(getRuntimeName()), any());
+  }
+
+  @Test
+  public void testCreateRuntime_defaultLabel_dataproc() throws ApiException {
+    when(userRuntimesApi.getRuntime(BILLING_PROJECT_ID, getRuntimeName()))
+        .thenThrow(new NotFoundException());
+    stubGetWorkspace(WORKSPACE_NS, WORKSPACE_ID, "test");
+
+    runtimeController.createRuntime(
+        BILLING_PROJECT_ID,
+        new Runtime().configurationType(RuntimeConfigurationType.DEFAULTDATAPROC));
+    verify(userRuntimesApi)
+        .createRuntime(
+            eq(BILLING_PROJECT_ID), eq(getRuntimeName()), createRuntimeRequestCaptor.capture());
+
+    LeonardoCreateRuntimeRequest createRuntimeRequest = createRuntimeRequestCaptor.getValue();
+    assertThat(((Map<String, String>) createRuntimeRequest.getLabels()).get("all-of-us-config"))
+        .isEqualTo("default-dataproc");
+  }
+
+  @Test
+  public void testCreateRuntime_defaultLabel_gce() throws ApiException {
+    when(userRuntimesApi.getRuntime(BILLING_PROJECT_ID, getRuntimeName()))
+        .thenThrow(new NotFoundException());
+    stubGetWorkspace(WORKSPACE_NS, WORKSPACE_ID, "test");
+
+    runtimeController.createRuntime(
+        BILLING_PROJECT_ID, new Runtime().configurationType(RuntimeConfigurationType.DEFAULTGCE));
+    verify(userRuntimesApi)
+        .createRuntime(
+            eq(BILLING_PROJECT_ID), eq(getRuntimeName()), createRuntimeRequestCaptor.capture());
+
+    LeonardoCreateRuntimeRequest createRuntimeRequest = createRuntimeRequestCaptor.getValue();
+    assertThat(((Map<String, String>) createRuntimeRequest.getLabels()).get("all-of-us-config"))
+        .isEqualTo("default-gce");
+  }
+
+  @Test
+  public void testCreateRuntime_overrideLabel() throws ApiException {
+    when(userRuntimesApi.getRuntime(BILLING_PROJECT_ID, getRuntimeName()))
+        .thenThrow(new NotFoundException());
+    stubGetWorkspace(WORKSPACE_NS, WORKSPACE_ID, "test");
+
+    runtimeController.createRuntime(
+        BILLING_PROJECT_ID, new Runtime().configurationType(RuntimeConfigurationType.USEROVERRIDE));
+    verify(userRuntimesApi)
+        .createRuntime(
+            eq(BILLING_PROJECT_ID), eq(getRuntimeName()), createRuntimeRequestCaptor.capture());
+
+    LeonardoCreateRuntimeRequest createRuntimeRequest = createRuntimeRequestCaptor.getValue();
+    assertThat(((Map<String, String>) createRuntimeRequest.getLabels()).get("all-of-us-config"))
```

Comment:
Would be good to validate that the machine type and disk are actually getting plumbed in a test case